File: hackerrank/30 days of code/day 10, binary numbers/binary_numbers.rb
# https://www.hackerrank.com/challenges/30-binary-numbers/problem?is_full_screen=true

def main
  n = gets.to_i
  binary = n.to_s(2)
  puts find_max_length_of_ones(binary)
end

def find_max_length_of_ones(binary)
  binary.split('0').map(&:length).max
end

main

File: hackerrank/30 days of code/day 10, binary numbers/binary_numbers.ts
// https://www.hackerrank.com/challenges/30-binary-numbers/problem?isFullScreen=true

'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString: string = '';
let inputLines: string[] = [];
let currentLine: number = 0;

process.stdin.on('data', function (inputStdin: string): void {
  inputString += inputStdin;
});

process.stdin.on('end', function (): void {
  inputLines = inputString.split('\n');
  inputString = '';
  main();
});

function readLine(): string {
  return inputLines[currentLine++];
}

//////////////////////////////////////////////////

function main() {
  let n: number = +readLine();
  let binary = n.toString(2);
  console.log(findMaxLengthOfOnes(binary));
}

function findMaxLengthOfOnes(binary: string): number {
  return Math.max(...binary.split('0').map(segment => segment.length));
}
